### PR TITLE
[Release] Use nightly Docker images

### PR DESCRIPTION
**Pull request by amogkam**
Anyscale now publishes nightly Ray Docker Images. We should use the nightly Ray docker image for our release tests instead of an outdated pinned nightly version. This ensures that we are testing Ray on an up to date Docker image.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by amogkam** [comment]
Release test results: https://buildkite.com/ray-project/periodic-ci/builds/1491, https://buildkite.com/ray-project/periodic-ci/builds/1489, https://buildkite.com/ray-project/periodic-ci/builds/1585

The failing tests are also failing on master or due to infra errors unrelated to these changes.

**Comment by rkooo567** [comment]
I am concerned about the reliability of nightly docker images. Is it guaranteed to be generated on every commit, as soon as new commit is merged?

File: release/nightly_tests/dataset/app_config.yaml [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 python:
-  pip_packages: []
+  pip_packages:
+    - boto3
+    - pyarrow
+    - tqdm
   conda_packages: []
 
 post_build_cmds:
-  - pip uninstall -y numpy ray || true
```
Comment (rkooo567) [inline release/nightly_tests/dataset/app_config.yaml:8]: Maybe keep ray uninstall?

Comment (amogkam) [reply release/nightly_tests/dataset/app_config.yaml:8]: Is it necessary? Tests look to be passing fine without it.

Comment (rkooo567) [reply release/nightly_tests/dataset/app_config.yaml:8]: I think when you provide the custom wheel url, this is needed iirc, -U doesn’t guarantee to redownload the wheel if the version is the same (which is the case for the custom url, both are mostly version 2.0)

Comment (amogkam) [reply release/nightly_tests/dataset/app_config.yaml:8]: Ah I see- I added it back to all of these.

**Review by rkooo567 — APPROVED** [review]

**Comment by gjoliver** [comment]
this is a really nice clean up change!

**Review by gjoliver — APPROVED** [review]

**Pull request merged**